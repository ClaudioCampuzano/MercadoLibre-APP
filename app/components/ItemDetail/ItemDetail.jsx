import React from 'react';
import styles from './ItemDetail.module.scss';
import Image from 'next/image';
import { formatCurrency } from '@/utils/formatCurrency';
const ItemDetail = ({ data }) => {
  const { title, description, picture, condition, sold_quantity: soldQuantity, price } = data;

  return (
    <article className={styles.article}>
      <section className={styles.section}>
        <Image src={picture} alt="article" width={400} height={400} />
        <div>
          <h4>{`${condition === 'new' ? 'Nuevo' : 'Usado'} ${soldQuantity} vendidos`}</h4>
          <h3>{title}</h3>
          <h1>{formatCurrency(price)}</h1>
          <button className={styles.button}>Comprar</button>
        </div>
      </section>
      <h2>Descripci√≥n del producto</h2>
      <p>{description}</p>
    </article>
  );
};

export { ItemDetail };
