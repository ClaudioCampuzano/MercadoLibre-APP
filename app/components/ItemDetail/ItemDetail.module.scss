@import '@/styles/constans.module.scss';

.article {
  width: 80%;
  padding: 15px;
  background-color: $whiteColor;
  box-shadow: 0 20px 25px -5px rgb(0 0 0 / 0.1), 0 8px 10px -6px rgb(0 0 0 / 0.1);
  border-radius: 0.125rem;
}

.imagenProduct {
  width: 50px;
  height: 50px;
}

.section {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}

.button {
  border-color: transparent;
  background-color: $secondaryColor;
  color: $whiteColor;
  width: 100%;
  height: 30px;
}
