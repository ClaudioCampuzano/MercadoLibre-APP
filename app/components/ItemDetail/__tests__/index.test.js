import React from 'react';

import { render } from '@testing-library/react';

import { ItemDetail } from '../ItemDetail';

jest.mock('next/navigation', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
    };
  },
}));

jest.mock('@/utils/formatCurrency', () => ({
  formatCurrency: () => '$456789',
}));

const data = {
  title: 'ipsum',
  description: 'lorem',
  picture: 'htt://blabla.com',
  condition: 'new',
  sold_quantity: 250,
  price: { amount: 15121, currency: 'CLP' },
};

const componentRender = (props = { data }) => {
  return render(<ItemDetail {...props} />);
};

describe('ItemDetail', () => {
  it('renders', () => {
    const { container } = componentRender();

    expect(container).toMatchSnapshot();
  });
});
