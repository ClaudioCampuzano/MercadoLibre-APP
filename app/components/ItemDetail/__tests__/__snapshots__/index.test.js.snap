// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetail renders 1`] = `
<div>
  <article
    class="article"
  >
    <section
      class="section"
    >
      <img
        alt="article"
        data-nimg="1"
        decoding="async"
        height="400"
        loading="lazy"
        src="/_next/image?url=htt%3A%2F%2Fblabla.com&w=828&q=75"
        srcset="/_next/image?url=htt%3A%2F%2Fblabla.com&w=640&q=75 1x, /_next/image?url=htt%3A%2F%2Fblabla.com&w=828&q=75 2x"
        style="color: transparent;"
        width="400"
      />
      <div>
        <h4>
          Nuevo 250 vendidos
        </h4>
        <h3>
          ipsum
        </h3>
        <h1>
          $456789
        </h1>
        <button
          class="button"
        >
          Comprar
        </button>
      </div>
    </section>
    <h2>
      Descripci√≥n del producto
    </h2>
    <p>
      lorem
    </p>
  </article>
</div>
`;
